packages/fast-usdc/src/util/cctp.js

```diff
@@ -0,0 +1,67 @@
+import { Buffer } from 'node:buffer';
+import { bech32 } from 'bech32';
+import { ethers } from 'ethers';
+
+/**
+ * Adapted from https://docs.noble.xyz/cctp/mint#encoding
+ *
+ * @param {string} address
+ * @returns {string}
+ */
+export const encodeBech32Address = address => {
+  const decoded = bech32.decode(address);
+  const rawBytes = Buffer.from(bech32.fromWords(decoded.words));
+
+  const padded = Buffer.alloc(32);
+  rawBytes.copy(padded, 32 - rawBytes.length);
+
+  return `0x${padded.toString('hex')}`;
+};
+
+const tokenAbi = ['function approve(address spender, uint256 value) external'];
+
+const contractAbi = [
+  'function depositForBurn(uint256 amount, uint32 destinationDomain, bytes32 mintRecipient, address burnToken) external',
+];
+
+export const makeProvider = (/** @type {string} */ rpc) =>
+  new ethers.JsonRpcProvider(rpc);
```

Comment:
> I guess by powerful you mean because it has network access?

yes. we only seem to use it as a fallback in cases where the caller can explicitly supply a provider, so I guess it doesn't do much harm. Want to add a comment to warn folks not to use it otherwise? (maybe not worth another round of ci. I won't insist)